storage_xml.c

```diff
@@ -120,29 +118,40 @@ static xt_status handle_account(struct xt_node *node, gpointer data)
 
 	if (!handle || !pass_b64 || !protocol || !prpl) {
 		return XT_ABORT;
-	} else if ((pass_len = base64_decode(pass_b64, (unsigned char **) &pass_cr)) &&
-	           arc_decode(pass_cr, pass_len, &password, xd->given_pass) >= 0) {
-		acc = account_add(xd->irc->b, prpl, handle, password);
-		if (server) {
-			set_setstr(&acc->set, "server", server);
-		}
-		if (autoconnect) {
-			set_setstr(&acc->set, "auto_connect", autoconnect);
-		}
-		if (tag) {
-			set_setstr(&acc->set, "tag", tag);
-		}
-		if (local) {
-			acc->flags |= ACC_FLAG_LOCAL;
-		}
-		if (locked && !g_strcasecmp(locked, "true")) {
-			acc->flags |= ACC_FLAG_LOCKED;
-		}
-	} else {
+	}
+
+	pass_len = base64_decode(pass_b64, (unsigned char **) &pass_cr);
+	if (pass_len < 0) {
```

Comment:
Checked glib just to be sure, it returns 0 and a possibly empty allocated string. You could probably remove the whole condition, or switch it to `==`. Whichever you prefer.